pydantic_ai_slim/pydantic_ai/prompt_config.py

```diff
@@ -0,0 +1,414 @@
+
+    If `None`, uses the default: 'Final result processed.'
+    """
+
+    output_tool_not_executed: str | Callable[[ToolReturnPart, RunContext[Any]], str] | None = None
+    """Message sent when an output tool call is skipped because a result was already found.
+
+    If `None`, uses the default: 'Output tool not used - a final result was already processed.'
+    """
+
+    output_validation_failed: str | Callable[[ToolReturnPart, RunContext[Any]], str] | None = None
+    """Message sent when an output tool fails validation."""
+
+    function_tool_not_executed: str | Callable[[ToolReturnPart, RunContext[Any]], str] | None = None
+    """Message sent when a function tool call is skipped because a result was already found.
+
+    If `None`, uses the default: 'Tool not executed - a final result was already processed.'
+    """
+
+    tool_denied: str | Callable[[ToolReturnPart, RunContext[Any]], str] | None = None
+    """Message sent when a tool call is denied by an approval handler.
+
+    If `None`, preserves the custom message from `ToolDenied` (or uses the default if none was set).
+    Set explicitly to override all denied tool messages.
+    """
+
+    validation_errors_retry_prompt: str | Callable[[RetryPromptPart, RunContext[Any]], str] | None = None
+    """Message appended to validation errors when asking the model to retry.
+
+    If `None`, uses the default: 'Fix the errors and try again.'
+    """
+
+    tool_retry_prompt: str | Callable[[RetryPromptPart, RunContext[Any]], str] | None = None
+    """Message appended when a `ModelRetry` exception is raised from a tool.
+
+    If `None`, uses the default: 'Fix the errors and try again.'
+    """
+
+    no_tool_retry_prompt: str | Callable[[RetryPromptPart, RunContext[Any]], str] | None = None
+    """Message appended when a `ModelRetry` exception is raised outside of a tool context.
+
+    If `None`, uses the default: 'Fix the errors and try again.'
+    """
+
+    prompted_output_template: str | None = None
+    """Template for prompted output schema instructions.
+
+    If set, this overrides the template used for [`PromptedOutput`][pydantic_ai.output.PromptedOutput].
+
+    If `None`, the template from `PromptedOutput` is used if set, otherwise the model profile default
+    ([`ModelProfile.prompted_output_template`][pydantic_ai.profiles.ModelProfile.prompted_output_template])
+    is used.
+    """
+
+    description_template: Callable[[str | list[pydantic_core.ErrorDetails]], str] | None = None
+    """Format a description message while asking the model to retry."""
+
+    def apply_template(self, message_part: ModelRequestPart, ctx: RunContext[Any]) -> ModelRequestPart:
+        if isinstance(message_part, ToolReturnPart):
+            if message_part.return_kind in (None, 'tool-executed'):
+                return message_part
+
+            field_name = message_part.return_kind.replace('-', '_')
+            template = getattr(self, field_name, None)
```

Comment:
I'd rather have a type safe method that directly maps from ToolReturnKind to the self.methods